eng/pipelines/docindex.yml

```
@@ -56,6 +69,11 @@ jobs:
           git status
         displayName: Checkout daily branch if it exists
         workingDirectory: $(DocRepoLocation)
+      - pwsh: |
+          $publicFeedUrl = "https://pkgs.dev.azure.com/azure-sdk/public/_packaging/azure-sdk-for-net/nuget/v3/index.json"
+          Write-Host "##vso[task.setvariable variable=PackageSourceOverride]$publicFeedUrl"
+        displayName: Set package source variable
+        workingDirectory: $(DocRepoLocation)
```

nit: spacing
```suggestion

      - pwsh: |
          $publicFeedUrl = "https://pkgs.dev.azure.com/azure-sdk/public/_packaging/azure-sdk-for-net/nuget/v3/index.json"
          Write-Host "##vso[task.setvariable variable=PackageSourceOverride]$publicFeedUrl"
        displayName: Set package source variable
        workingDirectory: $(DocRepoLocation)
        
```

Is there a reason this needs to be set as a variable dynamically here as opposed to hardcode below?

---

It is needed for toc generation. That's how we know whether to use public feeds for daily docs or not. 

Our common script doesn't provide the $PackageSourceOverride parameters, as other languages have their way to use this.
Java: we have a step to configure .m2 settings. [Link](https://github.com/Azure/azure-sdk-for-java/blob/main/eng/pipelines/docindex.yml#L96)

.NET has no way to pass this to the agent. So use variable for .NET only and read variable from script.

---

Here is the place to use the variable.
https://github.com/Azure/azure-sdk-for-net/pull/32062/files#r1008278055

---

Ok so is more about setting this as an env variable. Can you please add a comment to that effect here so we don't accidently remove it later and break that usage.